src/broadcast_socket.cpp

```diff
@@ -139,6 +139,32 @@ namespace libtorrent {
 #endif
 	}
 
+	TORRENT_EXTRA_EXPORT bool is_non_useful(address const& a)
+	{
+		TORRENT_TRY {
+#if TORRENT_USE_IPV6
+			error_code ec;
+			std::string s = a.to_string(ec);
```

Comment:
isn't there a way to do this without converting to a string? I'm a bit ignorant on IPv6 zones unfortunately.

Reply:
it was an easy way I found to get that suffix, do you think this is a costly operation in this context?